src/utils/version_helper.py

```diff
@@ -2,44 +2,50 @@
 
 
 class VersionHelper:
-    VERSION_NUMBER_PLUS_SIGN_REGEX = r"^(\d+)\.(\d+)\.(\d+)\+(\d+)$"
+    VERSION_REGEX = (
+        r'^'                  # Start of the string
+        r'(?:[a-zA-Z]+-)?'    # Optional ProjectName- prefix in a non-capturing group
+        r'\d+\.\d+\.\d+'      # Major.Minor.Patch version
+        r'(?:-M\d+)?'         # Optional -Mx suffix in a non-capturing group
+        r'[.+]'               # Separator, either . or +
+        r'(\d+)$'             # Build number in a captured group
+    )
 
     @staticmethod
-    def extract_build_number(version) -> int:
-        if not VersionHelper.is_valid_version(version):
-            raise ValueError(f'The provided version {version}  does not match the standardized format '
-                             f'used commonly across the organization: <MAJOR>.<MINOR>.<PATCH>.<BUILD NUMBER>')
-
-        parts = VersionHelper._sanitize_version(version).split('.')
-        if len(parts) != 4:
-            raise ValueError(f'The split version {version} must contains 4 parts')
-        return int(parts[3])
-
-    @staticmethod
-    def is_valid_sonar_version(version: str) -> bool:
-        if "." not in version:
-            return False
-        parts = version.split('.')
-        if len(parts) != 4:
-            return False
-
-        return True
-
-    @staticmethod
-    def is_valid_plus_signed_version(version: str) -> bool:
-
-        # This is an explicit requirement for project SLVSCODE
-        # see https://sonarsource.atlassian.net/browse/BUILD-4915 for more details
-
-        if not re.match(VersionHelper.VERSION_NUMBER_PLUS_SIGN_REGEX, version):
-            return False
-
-        return VersionHelper.is_valid_sonar_version(VersionHelper._sanitize_version(version))
-
-    @staticmethod
-    def is_valid_version(version: str) -> bool:
-        return VersionHelper.is_valid_sonar_version(version) or VersionHelper.is_valid_plus_signed_version(version)
+    def validate_version(version: str) -> None:
```

Comment:
You decided to remove `is_valid_plus_signed_version`, that's your right, however here we are loosing fundamental comments: WHY do we accept `+` -> 

```
     # This is an explicit requirement for project SLVSCODE
        # see https://sonarsource.atlassian.net/browse/BUILD-4915 for more details       
```

Such comments are literally gold, because when you refactor code, in order to simplify the program (part of clean code), you need these information.

Without such comments, you just check 3 projects (i.e: sonar-dummy sonar-dummy-oss well they don't use any `+` let's remove it it's making things complicated for nothing`) -> boom you accidentally broke a feature.

Or at very least a specific test should be there with these comments to say why we accept + and now that you add other variants, please add a ticket reference about WHY we support them.

This is very important to avoid future regression.

Reply:
Added the links back and done the same at https://github.com/SonarSource/gh-action_release/pull/217